group/views.py

```diff
@@ -20,140 +20,169 @@
+        raise Http404("Group does not exist!")
+    return group
+
+def get_running_contest(request, group_id):
+        
+    group = get_group(group_id)
+
+    all_contest = group.trace_contest.all()
+    all_running_contest_list = []
+    now = timezone.now()
+
+    for contest in all_contest:
+        if contest.start_time < now and contest.end_time > now:
+            all_running_contest_list.append(contest)
+
+    return render(
+        request, 'group/viewall.html', {
+            'data_list': all_running_contest_list, 
+            'title': 'running contest',
+            'list_type': 'runContest',
+        })
+
+def get_ended_contest(request, group_id):
+        
+    group = get_group(group_id)
+
+    all_contest = group.trace_contest.all()
+    all_ended_contest_list = []
+    now = timezone.now()
+
+    for contest in all_contest:
+        if contest.end_time < now:
+            all_ended_contest_list.append(contest)
+
+    return render(
+        request, 'group/viewall.html', {
+            'data_list': all_ended_contest_list, 
+            'title': 'ended contest',
+            'list_type': 'endContest',
+        })
+
+def get_all_announce(request, group_id):
+
+    group = get_group(group_id)
+
+    all_announce_list = group.announce.all()
+    return render(
+        request, 'group/viewall.html', {
+            'data_list': all_announce_list, 
+            'title': 'announce',
+            'list_type': 'announce',
+        })
 
     
-def detail(request,group_id):
-    running_contest_list = [
-        {'name':'11111', 's_time':'20111010', 'e_time':'20111010'},
-        {'name':'22222', 's_time':'20140505', 'e_time':'20111010'},
-        {'name':'33333', 's_time':'20122222', 'e_time':'20111010'},
-        {'name':'44444', 's_time':'20133333', 'e_time':'20111010'},
-        {'name':'55555', 's_time':'20144444', 'e_time':'20111010'},
-    ]
-
-    ended_contest_list = [
-        {'name':'aaaaa', 'time':'20111010'},
-        {'name':'bbbbb', 'time':'20140505'},
-        {'name':'ccccc', 'time':'20122222'},
-        {'name':'ddddd', 'time':'20133333'},
-        {'name':'eeeee', 'time':'20144444'},
-    ]
-
-    annowence_list = [
-        {'id':'1', 'name':'annowence_1', 'op':'drowsy', 'time':'20111010', 'content':'asdasd'},
-        {'id':'2', 'name':'annowence_2', 'op':'drowsy', 'time':'20140505', 'content':'qwefad'},
-    ]
-
-    student_list =[
-        {'name': 'Tom', 'time': '2014/1/2 23:00'},
-        {'name': 'Amy', 'time': '2014/1/2 23:00'},
-        {'name': 'Jess', 'time': '2014/1/2 23:00'},
-        {'name': 'Bieber', 'time': '2014/1/2 23:00'},
-        {'name': 'Ian', 'time': '2014/1/2 23:00'},
-        {'name': 'Anthony', 'time': '2014/1/2 23:00'},
-        {'name': 'Lily', 'time': '2014/1/2 23:00'}
-    ]
-
-    ta_list =[
-        {'name': 'drowsy'},
-        {'name': 'henry'},
-    ]
+def detail(request, group_id):
+    
+    group = get_group(group_id)
+
+    all_contest = group.trace_contest.all()[0:5]
```

Comment:
In my opinion, 5 may be replaced by some constant variable to improve readability.